### PR TITLE
Wire up MenuAnchor clipBehaviour property

### LINKED ISSUE #123628: MenuAnchor ```clipBehaviour``` property isn't wired up to the widget
Modifying the ```clipBehaviour``` property doesn't change the behaviour of the clip. 

https://user-images.githubusercontent.com/73116038/228339077-459b3adc-9b28-43c4-bd13-34b22d6a5b33.mp4

This is because even though the ```clipBehaviour``` property is passed down the tree, it isn't actually set.  On line 3315 of file menu_anchor.dart, the ```clipBehaviour``` property is hard coded as Clip.hardEdge.
This is occurring on both stable and master.

Will fill a pr shortly :)

### PR DESCRIPTION
**Pull request by whiskeyPeak**
Wires up the ```clipBehaviour``` property for the MenuAnchor widget. The default clip is Clip.hard as it was before, only now modifying this value actually leads to visual changes.

https://user-images.githubusercontent.com/73116038/228342040-aa4b59b4-e222-4abb-962a-bf2811cabe10.mp4

@gspencergoog I'm not really sure how to test this... Any thoughts?

Fixes #123628

### CODE REVIEW

**Comment by whiskeyPeak** [comment]
Checking the value of ```clipBehaviour``` wouldn't really test anything since the change in state is further down the tree and an entire golden test for this small bug fix seems a little excessive IMO

**Comment by whiskeyPeak** [comment]
I had to change an existing test because the "default" value for the ```SubMenuButton``` widget changed from ```Clip.none``` to ```Clip.hardEdge```. However, since the original default value was never used in the widget tree and a hardcoded value of ```Clip.hardEdge``` was used instead, there is no change in functionality. I hope this is ok? :)

**Comment by TahaTesser** [comment]
> @gspencergoog I'm not really sure how to test this... Any thoughts?

This requires a test in order to land. You can get the material from `_MenuPanel` and check its clip behavior (you can find similar test in the framework)

**Comment by whiskeyPeak** [comment]
Looks like the test failures from https://github.com/flutter/tests/pull/239 are back

File: packages/flutter/test/material/menu_anchor_test.dart [hunk 1]
```diff
@@ -945,6 +945,48 @@ void main() {
       expect(material.color, equals(Colors.red));
     });
 
+    testWidgets('make clipBehavior be set correctly for MenuAnchor', (WidgetTester tester) async {
```
Comment (TahaTesser) [inline packages/flutter/test/material/menu_anchor_test.dart:948]:
```suggestion
    testWidgets('MenuAnchor clip behavior', (WidgetTester tester) async {
```
Simply the title and make it similar to other clip behavior in the framework.

File: packages/flutter/test/material/menu_anchor_test.dart [hunk 2]
```diff
@@ -945,6 +945,48 @@ void main() {
...
+              ),
+            )
+          )
+        ),
+      );
+
+      await tester.tap(find.text('Tap me'));
+      await tester.pump();
+      final Material material = getMenuBarMaterial(tester);
+      expect(material.clipBehavior, equals(Clip.antiAlias));
```
Comment (TahaTesser) [inline packages/flutter/test/material/menu_anchor_test.dart:987]: Can you please test the default clip behavior first then the custom clip behavior

Also, please add test comments, something like:
```dart
// Test default clip behavior.
...


// Test custom clip behavior.
...
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/menu_anchor_test.dart (hunk 1 above)
Comment (whiskeyPeak) [reply packages/flutter/test/material/menu_anchor_test.dart]: Done

File: packages/flutter/test/material/menu_anchor_test.dart (hunk 2 above)
Comment (whiskeyPeak) [reply packages/flutter/test/material/menu_anchor_test.dart]: Done

Comment (TahaTesser) [reply packages/flutter/test/material/menu_anchor_test.dart]: This should be a single test. 

Firstly, don't pass any custom behavior, check the default clip behavior, follow by custom behavior, and check.

These two comments are in a single test. Sorry, if it wasn't clear.

Comment (whiskeyPeak) [reply packages/flutter/test/material/menu_anchor_test.dart]: I'm not sure how to do this with only a single widget. How do I set the property of a widget if I'm initially testing the default behaviour? Looked online but couldn't find anything helpful.

**Comment by whiskeyPeak** [comment]
@TahaTesser I think this is what you meant? :)

File: packages/flutter/test/material/menu_anchor_test.dart [hunk 3]
```diff
@@ -945,6 +945,91 @@ void main() {
...
+        ),
+      );
+
+      await tester.tap(find.text('Tap me'));
+      await tester.pump();
+      final Material material = getMenuBarMaterial(tester);
+      // Tests the provided MenuAnchor clip to be what is is set as
+      expect(material.clipBehavior, equals(Clip.antiAlias));
+    });
+
```
Comment (TahaTesser) [inline packages/flutter/test/material/menu_anchor_test.dart]:
```suggestion
    testWidgets('MenuAnchor clip behavior', (WidgetTester tester) async {
      await tester.pumpWidget(
        MaterialApp(
          home: Material(
            child: Center(
              child: MenuAnchor(
                menuChildren: const <Widget> [
                  MenuItemButton(
                    child: Text('Button 1'),
                  ),
                ],
                builder: (BuildContext context, MenuController controller, Widget? child) {
                  return FilledButton(
                    onPressed: () {
                      controller.open();
                    },
                    child: const Text('Tap me'),
                  );
                },
              ),
            )
          )
        )
      );

      await tester.tap(find.text('Tap me'));
      await tester.pump();

      // Test default clip behavior.
      expect(getMenuBarMaterial(tester).clipBehavior, equals(Clip.hardEdge));

      // Close the menu.
      await tester.tapAt(const Offset(10.0, 10.0));
      await tester.pumpAndSettle();

      await tester.pumpWidget(
        MaterialApp(
          home: Material(
            child: Center(
              child: MenuAnchor(
                clipBehavior: Clip.antiAlias,
                menuChildren: const <Widget> [
                  MenuItemButton(
                    child: Text('Button 1'),
                  ),
                ],
                builder: (BuildContext context, MenuController controller, Widget? child) {
                  return FilledButton(
                    onPressed: () {
                      controller.open();
                    },
                    child: const Text('Tap me'),
                  );
                },
              ),
            )
          )
        )
      );

      await tester.tap(find.text('Tap me'));
      await tester.pump();

      // Test custom clip behavior.
      expect(getMenuBarMaterial(tester).clipBehavior, equals(Clip.antiAlias));
    });

```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/menu_anchor_test.dart (hunk 3 above)
Comment (whiskeyPeak) [reply packages/flutter/test/material/menu_anchor_test.dart]: Done :+1:

**Review by TahaTesser — APPROVED** [review]
LGTM

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Thanks for catching (and fixing!) that.

**Pull request merged**